ci/new_tsqa/tests/test_spdy_protocol_select.py

```diff
@@ -0,0 +1,141 @@
+        build_dir = os.environ.get('top_builddir', '../..')
+        cls.client = '%s/spdylay/src/spdycat' % build_dir
+        log.info('top build_dir = {0}'.format(build_dir))
+        log.info('spdycat path = {0}'.format(cls.client))
+
+        # get spdy server ports
+        cls.spdy_port = tsqa.utils.bind_unused_port()[1]
+        log.info('spdy server port = {0}'.format(cls.spdy_port))
+        cls.http_port = tsqa.utils.bind_unused_port()[1]
+        log.info('http server port = {0}'.format(cls.http_port))
+
+        cls.configs['remap.config'].add_line('map / http://www.yahoo.com/\n')
+        
+        # set only one ET_NET thread (so we don't have to worry about the per-thread pools causing issues)
+        cls.configs['records.config']['CONFIG']['proxy.config.exec_thread.limit'] = 1
+        cls.configs['records.config']['CONFIG']['proxy.config.exec_thread.autoconfig'] = 0
+ 
+        # SPDY configs
+        cls.configs['records.config']['CONFIG']['proxy.config.http.server_ports'] += ' {0}:ssl {1}:proto=http:ssl'.format(cls.spdy_port, cls.http_port)
+        cls.configs['records.config']['CONFIG']['proxy.config.ssl.server.cert.path'] = helpers.tests_file_path('rsa_keys')
+        
+        # optional enable debug logs
+        #cls.configs['records.config']['CONFIG']['proxy.config.diags.debug.enabled'] = 1
+        #cls.configs['records.config']['CONFIG']['proxy.config.diags.debug.tags'] = 'http.*|dns.*|ssl'
+
+        # configure SSL multicert
+        cls.configs['ssl_multicert.config'].add_line('dest_ip=* ssl_cert_name={0}\n'.format(helpers.tests_file_path('rsa_keys/www.example.com.pem')))
+
+    @classmethod
+    def callSpdycat(self, port, path, args):
+      full_args = [self.client,'https://localhost:%d%s' % (port, path)] + args
+      self.log.info('full args = {0}'.format(full_args))
+      p = subprocess.Popen(full_args, stdout=subprocess.PIPE,
+          stdin=subprocess.PIPE)
+      self.stdout, self.stderr = p.communicate()
+      return p.returncode
+
+"""
+TODO: re-add spdy2 tests. looks like support here might be lacking some way. was not able to get ATS to advertise spdy/2
+even when it was explicitly set with proto=spdy/2
+"""
+class TestSPDYv2(TestSPDY):
+    @classmethod
+    def setUpClass(cls):
+        '''
+        Skip spdy2 tests for now
+        '''
+        raise helpers.unittest.SkipTest('Skipping spdy/2 tests')
+    
+    @classmethod
+    def setUpEnv(cls, env):
+        '''
+        This function is responsible for setting up the environment for this fixture
+        This includes everything pre-daemon start
+        '''
+        
+        cls.spdy2_port = tsqa.utils.bind_unused_port()[1]
+        log.info('spdy2 server port = {0}'.format(cls.spdy2_port))
+        # make sure we add port supports spdy2
+        cls.configs['records.config']['CONFIG']['proxy.config.http.server_ports'] += ' {0}:proto=spdy/2:ssl'.format(cls.spdy2_port)
+
+        super(TestSPDYv2, cls).setUpEnv(env)
```

Comment:
This should be called before you configure the spdy2 port-- the way this is right now your config changes will be overwritten by the ones in TestSpdy.setUpEnv()